src/services/storage/index.ts

```diff
@@ -37,6 +41,33 @@ export class StakeService extends Service {
   emit?: Function
 }
 
+export class AvgBillingPriceService extends Service {
+  emit?: Function
+
+  async get (minMax: MinMax): Promise<number | string> {
+    if (!config.get('storage.tokens')) {
+      throw new Error('"storage.tokens" not exist in config')
+    }
+    const sequelize = this.Model.sequelize
+
+    const getAvgMinMaxBillingPrice = (minMax: MinMax): string => `
+        SELECT
+            CAST(
+              (SUM(price * COALESCE("rates"."usd", 0)) / COUNT(*) * 1024 / period * (3600 * 24)) as INTEGER
+            ) as avgPrice
+        FROM "storage_billing-plan"
+        LEFT OUTER JOIN
+            "rates" AS "rates" ON "storage_billing-plan"."rateId" = "rates"."token"
+        GROUP BY offerId
+        ORDER BY avgPrice ${minMax === -1 ? 'DESC' : 'ASC'}
+        LIMIT 1
+     `
+    const [{ avgPrice }] = await sequelize.query(getAvgMinMaxBillingPrice(minMax), { type: QueryTypes.SELECT, raw: true })
+    // Maybe not the best idea
+    return minMax === -1 ? avgPrice + 10 : avgPrice - 10
```

Comment:
Why `+- 10`?

Reply:
the problem is that our token rates are dynamic and the UI query min/max values for filters. Then in a few minutes/seconds after the rates table updated the min/max values are changed and then we miss some of the offers because of that.
I know that's is not a good idea, but nothing comes to my mind

Reply:
Removed